### DISCUSSION TITLE
Improve default DragPreview mouse coordinates

### DISCUSSION THREAD

**Discussion by AlexGalays** [adobe/react-spectrum#4801] [Ideas]
What do you think about the below improvement at: 

https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/dnd/src/useDrag.ts#L140C1-L143C10

instead:

```ts
if (x > size.width) x = (x * size.width) / rect.width
if (y > size.height) y = (y * size.height) / rect.height
```

This fixes two things:

1) The preview height can be smaller than the original item height but have the same width (or vice versa): the new code will keep an axis coordinate if it fits. Aggressively defaulting to the middle is very strange when you initiate the drag from the right hand side, just because ONE dimension doesn't fit.

2) It feels way more natural to me to default to the same general area/proportions of where the drag was initiated.

**Comment by reidbarber** [comment]
This seems like an improvement to me, feel free to make a PR.
